sklearn/model_selection/tests/test_split.py

```diff
@@ -1283,6 +1283,17 @@ def test_train_test_split():
         train, test = train_test_split(y, shuffle=False, test_size=test_size)
         assert_array_equal(test, [8, 9])
         assert_array_equal(train, [0, 1, 2, 3, 4, 5, 6, 7])
+    
+    # Based on issue #20774 - check for integer overflow on 32-bit platforms
```

Comment:
Could you put this part in a separate function for testing?

Reply:
Done

Reply:
To follow the template for non-regression tests.
```suggestion
    """Check for integer overflow on 32-bit platforms.

    Non-regression test for:
    https://github.com/scikit-learn/scikit-learn/issues/20774
    """
```